degirum_tools/regression_eval.py

```diff
@@ -0,0 +1,111 @@
+#
+# regression_eval.py: evaluation toolkit for regression models used in PySDK samples
+#
+# Copyright DeGirum Corporation 2023
+# All rights reserved
+#
+
+import yaml
+import json
+import os
+import numpy as np
+
+
+class ImageRegressionModelEvaluator:
+    def __init__(
+        self,
+        dg_model,
+        input_resize_method="bilinear",
+        input_pad_method="crop-first",
+        image_backend="opencv",
+        input_img_fmt="JPEG",
+        input_letterbox_fill_color=(114, 114, 114),
+        input_numpy_colorspace="auto",
+    ):
+        """
+        Constructor.
+            This class evaluates the MAE and MSE for Image Regression models.
+
+            Args:
+                dg_model (Regression model): Regression model from the DeGirum model zoo.
+                input_resize_method (str): Input Resize Method.
+                input_pad_method (str): Input Pad Method.
+                image_backend (str): Image Backend.
+                input_img_fmt (str): InputImgFmt.
+                input_letterbox_fill_color (tuple): the RGB color for padding used in letterbox
+                input_numpy_colorspace (str): input colorspace: ("BGR" to match OpenCV image backend)
+        """
+
+        self.dg_model = dg_model
+        self.dg_model.input_resize_method = input_resize_method
+        self.dg_model.input_pad_method = input_pad_method
+        self.dg_model.image_backend = image_backend
+        self.dg_model.input_image_format = input_img_fmt
+        self.dg_model.input_numpy_colorspace = input_numpy_colorspace
+        self.dg_model.input_letterbox_fill_color = input_letterbox_fill_color
+
+    @classmethod
+    def init_from_yaml(cls, dg_model, config_yaml):
+        """
+        config_yaml (str) : Path of the yaml file that contains all the arguments.
+
+        """
+        with open(config_yaml) as f:
+            args = yaml.load(f, Loader=yaml.FullLoader)
+
+        return cls(
+            dg_model=dg_model,
+            input_resize_method=args["input_resize_method"],
+            input_pad_method=args["input_pad_method"],
+            image_backend=args["image_backend"],
+            input_img_fmt=args["input_img_fmt"],
+            input_letterbox_fill_color=tuple(args["input_letterbox_fill_color"]),
+            input_numpy_colorspace=args["input_numpy_colorspace"],
+        )
+
+    def evaluate(
+        self,
+        image_folder_path: str,
+        ground_truth_annotations_path: str,
+        num_val_images: int = 0,
+        print_frequency: int = 0,
+    ):
```

Comment:
please annotate function return: `Tuple(float, float)` I guess?
Note: we should be Python 3.8 compatible, so not `tuple[float, float]`, but `Tuple(float, float)`